code/game/g_weapon.c

```diff
@@ -104,55 +105,61 @@ static void WP_FireHyperspanner(gentity_t *ent, qboolean alt_fire) {
     gentity_t*	nearest = NULL;
     int			count = 0;
     float		nearestd = 65000;
-    vec3_t		dVec, end;
-    vec3_t		mins = { -40, -40, 0 }, maxs = { 40, 40, 0 };
     struct list	classnames;
 
+    int radius = 96;
+
     /* prepare lists */
     list_init(&classnames, free);
     list_init(&validEnts, free);
     classnames.append(&classnames, "func_breakable", LT_STRING, strlen("func_breakable") + 1);
     classnames.append(&classnames, "misc_model_breakable", LT_STRING, strlen("misc_model_breakable") + 1);
 
     /* find all vlaid entities in range */
-    count = G_RadiusListOfTypes(&classnames, ent->r.currentOrigin, 512, NULL, &validEnts);
+    count = G_RadiusListOfTypes(&classnames, ent->r.currentOrigin, radius, NULL, &validEnts);
     classnames.clear(&classnames);
     //G_Printf("Found %d possible candidates\n", count);
 
     if (count <= 0){
         return;
     }
 
-    trace_t tr;
-
     iter = validEnts.iterator(&validEnts, LIST_FRONT);
     for (cont = validEnts.next(iter); cont != NULL; cont = validEnts.next(iter)) {
         e = cont->data;
 
-        // TODO: fix problems with small distance
-        if (e->spawnflags & 512) {
-            VectorSubtract(ent->r.currentOrigin, e->s.angles2, dVec);
-            VectorMA(e->s.angles2, 1024, dVec, end);
-            trap_Trace(&tr, e->s.angles2, mins, maxs, end, e->s.number, MASK_SHOT);
-        } else {
-            VectorSubtract(ent->r.currentOrigin, e->s.origin, dVec);
-            VectorMA(e->s.origin, 1024, dVec, end);
-            trap_Trace(&tr, e->s.origin, mins, maxs, end, e->s.number, MASK_SHOT);
-        }
-        //G_Printf("Checking entity: %d\n", i);
-        if (tr.entityNum != ent->s.number) {
+        if (e->health >= e->damage) {
+            // If it ain't broke, don't fix it.
             continue;
         }
-        //G_Printf("Nothing is blocking view ...\n");
+
+        vec3_t distanceVec;
         if (e->spawnflags & 512) {
```

Comment:
What flag is 512? Can we put a name to it, with some constant/define/etc. ? (I realise this isn't new, but since we're here...)